File: app_launcher/src/main/java/com/pandaq/app_launcher/ui/framework/AppBaseFragment.java
package com.pandaq.app_launcher.ui.framework;

import com.pandaq.appcore.framework.mvp.BaseFragment;
import com.pandaq.appcore.framework.mvp.BasePresenter;
import com.pandaq.uires.progress.LoadingDialog;

/**
 * Created by huxinyu on 2018/1/26.
 * Email : panda.h@foxmail.com
 * Description :BaseFragment 基类
 */

public abstract class AppBaseFragment<P extends BasePresenter> extends BaseFragment<P> {

    @Override
    public void onPause() {
        super.onPause();
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
    }

    @Override
    public void showLoading(String msg) {
        if (getActivity() instanceof AppBaseActivity) {
            ((AppBaseActivity) getActivity()).showLoading(msg);
        }
    }

    @Override
    public void hideLoading() {
        if (getActivity() instanceof AppBaseActivity) {
            ((AppBaseActivity) getActivity()).hideLoading();
        }
    }

    @Override
    public void onError(long errCode, String errMsg) {

    }

    @Override
    public void onLoadFinish() {

    }
}

File: app_launcher/src/main/java/com/pandaq/app_launcher/ui/framework/AppBaseFragment.kt
package com.pandaq.app_launcher.ui.framework

import com.pandaq.appcore.framework.app.ActivityTask
import com.pandaq.appcore.framework.mvp.BaseFragment
import com.pandaq.uires.loading.LoadingDialogUtil
import com.pandaq.uires.msgwindow.Toaster

/**
 * Created by huxinyu on 2019/7/8.
 * Email : panda.h@foxmail.com
 * <p>
 * Description :
 */
abstract class AppBaseFragment<P : AppBasePresenter<*>> : BaseFragment<P>() {
    override fun showLoading(msg: String?) {
        LoadingDialogUtil.show(ActivityTask.getInstance().topActivity, msg)
    }

    override fun showLoading() {
        showLoading(true)
    }

    override fun showLoading(cancelAble: Boolean) {
        LoadingDialogUtil.show(ActivityTask.getInstance().topActivity, cancelAble)
    }

    override fun hideLoading() {
        LoadingDialogUtil.hideProgressQuick()
    }

    override fun onError(errCode: Long, errMsg: String?) {
        errMsg?.let {
            Toaster.msg(errMsg).showError()
        }
    }

    override fun onLoadFinish(success: Boolean) {
        hideLoading()
    }
}